tests/test_buffer.py

```diff
@@ -222,6 +222,40 @@ def test_SimpleArray_clone(self):
         self.assertEqual(sarr_ref[3], 3.0)
         self.assertEqual(sarr_clone[3], 2.0)  # should be the original value
 
+    def test_SimpleArray_transpose(self):
+        def check(sarr, ndarr):
+            self.assertEqual(sarr.shape, ndarr.shape)
+            shape = sarr.shape
+            for idx1 in range(shape[0]):
+                for idx2 in range(shape[1]):
+                    for idx3 in range(shape[2]):
+                        for idx4 in range(shape[3]):
+                            ndnum = ndarr[idx1, idx2, idx3, idx4]
+                            sarrnum = sarr[idx1, idx2, idx3, idx4]
+                            self.assertEqual(ndnum, sarrnum)
+
+        ndarr = np.arange(2 * 3 * 4 * 5, dtype='float64')
+        ndarr = ndarr.reshape((2, 3, 4, 5))
+        ndarrT = ndarr.transpose()
+
+        sarr = modmesh.SimpleArrayFloat64(array=ndarr)
+        sarr.transpose()
+        check(sarr, ndarrT)
```

Comment:
Could you please also test `sarr.transpose().copy()`?

Reply:
This will be an issue.  In that case, there will not be an equivalent of `ndarr.transpose.copy()`.

Let's give it more discussions and revise in a new PR.